ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -1604,12 +1606,14 @@ private void finalizeMetadataChunk() {
               passedInBlobProperties.isEncrypted(), passedInBlobProperties.getExternalAssetTag());
       if (isStitchOperation() || getNumDataChunks() > 1) {
         // values returned are in the right order as TreeMap returns them in key-order.
-        List<StoreKey> orderedChunkIdList = new ArrayList<>(indexToChunkIds.values());
-        buf = MetadataContentSerDe.serializeMetadataContent(intermediateChunkSize, getBlobSize(), orderedChunkIdList);
+        List<Pair<StoreKey, Long>> orderedChunkIdList = indexToChunkIds.values().stream().collect(
+            Collectors.toList());
+
+        buf = MetadataContentSerDe.serializeMetadataContentV3(getBlobSize(), orderedChunkIdList);
```

Comment:
We don't want to start writing in v3 before all frontends have been deployed with read support for metadata content v3. 

Perhaps we should add a router config for the metadata content version like we have for blob id version and then pass the version number into the `MetadataContentSerDe.serializeMetadataContent()` method

Reply:
added config